crates/isograph_compiler/src/write_artifacts.rs

```diff
@@ -1,17 +1,21 @@
 use std::{
+    collections::BTreeMap,
     fs::{self, File},
     io::Write,
     path::PathBuf,
 };
 
-use common_lang_types::ArtifactPathAndContent;
+use common_lang_types::{ArtifactPathAndContent, DocumentId, QueryText};
 use intern::string_key::Lookup;
+use isograph_config::CompilerConfig;
 use thiserror::Error;
 
 pub(crate) fn write_artifacts_to_disk(
     paths_and_contents: impl IntoIterator<Item = ArtifactPathAndContent>,
```

Comment:
Can we keep this file unchanged, and make `path_and_contents` contain the persisted documents JSON if necessary?

Reply:
Yes, but only if we agreed on generating json inside artifact_directory. I think we can do it and refactor it later if necessary